src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkConfig.java

```diff
@@ -337,6 +365,14 @@ public String getAuthAppkey() {
     public String getAuthAuthority() {
         return this.getString(KUSTO_AUTH_AUTHORITY_CONF);
     }
+
+    public boolean getManagedIdentityEnabled() {
```

Comment:
Just a quick question , should we add a validation where MSI is enabled but client id is not passed ?

Reply:
I actually had to change my first version. Usually, the clientId is optional when using managed identities because the "execution context" allows to deduce it. For example, if the managed identity authentication is done in a pod inside AKS, and [AAD Pod Identity](https://github.com/Azure/aad-pod-identity) is properly setup, then the clientId is deduced from the Pod.

After rebasing on the latest version of master, we do need both the Client ID and the Tenant ID in order to validate the accesses on the table. This is a little bit more involved but that's fine.

Now, the AppId and AuthAuthority are required in both managed identity and AAD Application use cases, the password is only needed when using an AAD Application.

Reply:
You can see that the access validation is optional in the newest version - it would actually be turned off by default in next major version. Regardless clientId is optional, I think its only required for user assigned - where do you see required?
If you do manage to use it without clientId and authority id you can make both optional as well

Reply:
BTW it seems like it should just work out of the box - but did you try it with distributed kafka?

Reply:
> You can see that the access validation is optional in the newest version - it would actually be turned off by default in next major version.

Indeed. Well the clientId would only be required if not using managed identities _or_ if validation of permissions is needed.

> Regardless clientId is optional, I think its only required for user assigned - where do you see required?

It's globally rather optional, usually the VM doing a call to IMDS is enough to deduce the managed identity used to authenticate, regardless of the fact that it's a system or user assigned identity.

There are some libraries that are less flexible and that would require the clientId set explicitly (not the case here) _or_ when using multiple managed identity in the same VM / pod then it's necessary to specify the clientId to use (potentially the case here).

> If you do manage to use it without clientId and authority id you can make both optional as well

In my use case (Kafka Connect deployed in a AKS cluster with AAD Pod Identity installed and correctly configured), if validation is turned off, both these informations are not needed and can be deduced from the context of the call.

> BTW it seems like it should just work out of the box - but did you try it with distributed kafka?

I'm not sure what you mean, but if we are talking about a cluster of Kafka Connect workers (as opposed to a standalone instance), then it works out of the box indeed.